crates/core/src/otel.rs

```diff
@@ -23,6 +30,86 @@ pub struct OtelConfig {
     pub metrics_endpoint: Option<String>,
     /// Overrides the OpenTelemetry endpoint for logs.
     pub logs_endpoint: Option<String>,
+    /// Determines whether grpc or http will be used for exporting the telemetry.
+    pub protocol: Option<String>,
+}
+
+impl OtelConfig {
+    pub fn parse(self) -> anyhow::Result<Self> {
+        match self.protocol.clone().unwrap_or_default().as_ref() {
+            "http" | "grpc" => {}
+            protocol => bail!("unsupported protocol: {:?}", protocol),
+        };
+        Ok(self)
+    }
+
+    pub fn otel_protocol(&self) -> OtelProtocol {
+        match self
+            .protocol
+            .clone()
+            .unwrap_or(DEFAULT_PROTOCOL.to_string())
+            .as_str()
```

Comment:
Clippy is probably gonna get you here with the `unwrap_or` function call instead of `unwrap_or_else`, but if you're able to change the data type of the protocol this function can be a bit simpler

Reply:
I was struggling with this, turns out what I really wanted was the `impl FromStr for OtelProtocol` which essentially solved this the way I wanted in the first place.